src/components/Input.tsx

```diff
@@ -1,129 +1,116 @@
 
-    const composedStyles = useMemo(
-      () =>
-        StyleSheet.compose<ViewStyle>(
-          [
-            fullWidth && s.wFull,
-            s.fontLato,
-            s.itemsCenter,
-            s.itemsCenter,
-            isPasswordType && s.justifyBetween,
-            s.minW48,
-            s.pR2,
-            s.h10,
-            s.roundedSm,
-            editable && s.border,
-            s.borderGray300,
-            isFocused && [s.borderSecondaryDark, s.inputShadow],
-            error && s.borderError,
-          ],
-          style
-        ),
-      [isPasswordType, fullWidth, s, editable, isFocused, error, style]
-    )
-
     return (
-      <View style={inputContainerStyle}>
-        {label ? (
-          <Text.CaptionBold color={error ? 'error' : 'secondaryDark'} style={[s.selfStart, s.mB1]}>
+      <FormControl isRequired={isRequired} isInvalid={Boolean(errorMessage)} {...layoutProps}>
+        {label && (
+          <FormControl.Label
+            _disabled={{
+              _text: {
+                color: 'gray.400',
+                fontWeight: 'bold',
+              },
+            }}
+          >
             {label}
-          </Text.CaptionBold>
-        ) : null}
-        <Box
-          alignItems="center"
-          flexDirection="row"
-          bg="white"
-          style={composedStyles}
-          bgOpacity={editable ? 0.8 : 1}
-        >
-          <TextInput
-            placeholder={placeholder}
-            autoCapitalize="none"
-            editable={editable}
-            ref={ref}
-            secureTextEntry={securePassword}
-            onFocus={handleFocus}
-            style={[
-              s.pX4,
-              s.flex1,
-              s.textBase,
-              s.textSecondaryDark,
-              !editable && s.textGray300,
-              isFocused && s.textSecondaryDark,
-              error && s.textError,
-            ]}
-            onBlur={handleBlur}
-            placeholderTextColor={colors.gray300}
-            testID="MAIN_INPUT"
-            value={isEmailType ? value?.trim() : value}
-            {...props}
-          />
-          {isPasswordType && (
-            <Touchable onPress={toggleSecurePassword}>
-              <Feather name={iconName} color={colors.gray300} size={24} />
-            </Touchable>
-          )}
-        </Box>
-      </View>
+          </FormControl.Label>
+        )}
+        <NBInput
+          {...inputProps}
+          onBlur={handleBlur}
+          ref={ref || undefined}
+          secureTextEntry={securePassword}
+          rightElement={
+            inputProps.type === 'password' ? (
+              <Touchable mr={2} onPress={toggleSecurePassword}>
+                <Feather name={iconName} color={colors.gray['400']} size={24} />
+              </Touchable>
+            ) : (
+              inputProps.rightElement
+            )
+          }
+        />
+        {helperText && <FormControl.HelperText>{helperText}</FormControl.HelperText>}
```

Comment:
`HelperText` could be wrapped with our own component just to apply styles, the same goes with `ErrorMessage`.

What I'd like us to do is to go with the most simple abstraction up to the more advanced use cases, so:

- basic atoms (Input, Checkbox, HelperText, ErrorMessage, etc
- molecule - Field, FieldController, etc,